sycl/include/sycl/ext/intel/experimental/esimd/detail/test_proxy.hpp

```diff
@@ -0,0 +1,106 @@
+//==-------------- test_proxy.hpp - DPC++ Explicit SIMD API ----------------==//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+// Test proxy to differentiate move and copy constructor calls
+//===----------------------------------------------------------------------===//
+
+#pragma once
+
+#ifndef __ESIMD_ENABLE_TEST_PROXY
```

Comment:
It's expected for the proxy class to be available in device code, so that it could be incorporated into the ESIMD API classes. See `__ESIMD_DECLARE_TEST_PROXY_ACCESS` usage.
I've added a comment to the code, please let me know if there are any other concerns